packages/app/src/app/overmind/namespaces/dashboard/actions.ts

```diff
@@ -1972,18 +1972,65 @@ export const getRepositoriesByTeam = async ({ state, effects }: Context) => {
   try {
     dashboard.repositories = null;
 
-    const repositoriesData = await effects.gql.queries.getRepositoriesByTeam({
-      teamId: activeTeam,
+    // First fetch data without syncing with GitHub
+    // to decrease waiting time.
+    const unsyncedRepositoriesData = await effects.gql.queries.getRepositoriesByTeam(
+      {
+        teamId: activeTeam,
+        syncData: false,
+      }
+    );
+    const unsyncedRepositories = unsyncedRepositoriesData?.me?.team?.projects;
+    if (!unsyncedRepositories?.length) {
+      return;
+    }
+
+    dashboard.repositories = unsyncedRepositories;
+
+    // Then fetch data synced with GitHub to make sure
+    // what we show is up-to-date.
+    const syncedRepositoriesData = await effects.gql.queries.getRepositoriesByTeam(
```

Comment:
I wonder if we should extract this into a separate function that runs when the page becomes idle, so it doesn't affect the first contentful paint. Not sure if this might be an issue for now, as it's just a async call that should at most trigger one re-render in the react hierarchy. Just a passing thought.

Reply:
@alexnm interesting, your idea is for the repositories page to trigger another action after getting the initial data? let's follow up in slack/linear.